### DISCUSSION TITLE
Rules - Power on if sensor 1 unter 7, save value of sensor2, power off when sensor 2 reaches that value again

### DISCUSSION THREAD

**Discussion by AndaSquared** [arendst/Tasmota#14452] [Q&A] [answered]
Hi all, 
thank you so much for providing us all the cool features in Tasmota.
I am quite new into this – just used it for SonOff devices before.

Right now, I am starting a new project and want to implement some rules.
I already read the docs and basically understand the rules, but I did not yet find a solution for my case.

I have controlled domestic ventilation with a geothermal energy preheating module, which is really stupid controlled by the manufacturer. I want to control the pump by myself with Tasmota.
For that, I have added many sensors to the different positions in the airflow.

My goal is to do following (in the winter):
•	Start the pump when the outgoing air is below 7 degrees
•	Save the temperature of the ingoing temperature at that moment (lets say var1) when power1 = on
•	Stop the pump when the temperature of the ingoing temperature reaches the value of var1 again

The same also for the summer:
•	Start the pump when the outgoing air is higher than 27 degrees
•	Save the temperature of the ingoing temperature (once)
•	Stop the pump when the, when the ingoing temperature drops under var 1 again.

Did any of you do something like that?
For me the problem is the combination of once / every time. I don’t know If it is possible to do this inside one rule.
As I am quite new to rules in Tasmota, I maybe also think completely wrong.

Would be happy if somebody could help me out here.

Thank you very much
AndaSquared

**Comment by AndaSquared** [comment]
here is a graph which could explain it a bit better:

![image](https://user-images.githubusercontent.com/97802343/149632940-158107f9-4b2c-4dc9-86cc-0945fb184d63.png)

**Comment by barbudor** [comment, marked as answer]
The graph doesn't explain better as I don't understand the legend
But assuming that switching the pump on/off is doing what you expect and based on your description, the below would do what you wrote.
Always as this one enabled to continuously save the value of Ingoing temperature into var2
```
Rule1
    ON Ingoing#Temperature DO var2 %value% ENDON
Rule1 1
```
For the winter:
```
Rule2
    ON Outgoing#Temperature<7 DO backlog Power ON; var1 %var2% ENDON
    ON Ingoing#Temperature<%var1% DO Power OFF ENDON
Rule2 1
Rule2 5
```
For the summer
```
Rule3
    ON Outgoing#Temperature>27 DO backlog Power ON; var1 %var2% ENDON
    ON Ingoing#Temperature<%var1% DO Power OFF ENDON
Rule3 1
Rule3 5
```
You will have to enable/disable Rule2/3 depending on the season
Replace Ingoing and Outgoing by your sensor name

**Comment by AndaSquared** [comment]
Hi barbudor, 

Thank you very much for your rules.
that one did the trick

ON event#t2<7 DO backlog Power ON; **var1 %var2%** ENDON

Now this one is in use and it looks good.


    Rule1
    ON DS18B20-1#Temperature DO event t1=%value% ENDON
	ON DS18B20-6#Temperature DO event t2=%value% ENDON
	ON event#t1 DO var2 %value% ENDON
    Rule1 1

    Rule2
    ON event#t2<7 DO backlog Power ON; var1 %var2% ENDON
    ON event#t1>%var1% DO Power OFF ENDON
    Rule2 1
    Rule2 5

    Rule3
    ON event#t2>27 DO backlog Power ON; var1 %var2% ENDON
    ON event#t1<%var1% DO Power OFF ENDON
    Rule3 1
    Rule3 5


thx
AndaSquared

**Reply by barbudor** [reply]
Using events doesn't bring any value.
You could use the DS18XB20 trigger everywhere directly

**Reply by AndaSquared** [reply]
Hi barbudor,

your are right, I know.
I added this to be able to test it more easily

For example with:

Backlog event t1=-4;event t2=8
Backlog event t1=-6;event t2=6
Backlog event t1=-6;event t2=16
Backlog event t1=-5;event t2=16
Backlog event t1=-4;event t2=16
Backlog event t1=-3;event t2=16

So I could test the triggers without waiting for the temperature to change so the rules would switch the relay

Thank you, 
Regards
AndaSquared
